pkg/helpers/identifier_string_parser_test.go

```diff
@@ -91,4 +94,251 @@ func Test_ParseIdentifierString(t *testing.T) {
 		require.NoError(t, err)
 		containsAll(t, parts, expected)
 	})
+
+	t.Run("empty identifier", func(t *testing.T) {
+		input := `""`
+		expected := []string{""}
+
+		parts, err := ParseIdentifierString(input)
+
+		require.NoError(t, err)
+		containsAll(t, parts, expected)
+	})
+
+	t.Run("handled correctly double quotes", func(t *testing.T) {
+		input := `""."."".".".""."".""."".""."".""."""""`
+		expected := []string{"", `.".`, `.".".".".".".".""`}
+
+		parts, err := ParseIdentifierString(input)
+
+		require.NoError(t, err)
+		containsAll(t, parts, expected)
+	})
+}
+
+func Test_ParseAccountObjectIdentifier(t *testing.T) {
+	testCases := []struct {
+		Input    string
+		Expected sdk.AccountObjectIdentifier
+		Error    string
+	}{
+		{Input: ``, Error: "incompatible identifier: "},
+		{Input: "a\nb", Error: "incompatible identifier: a\nb"},
+		{Input: `a"b`, Error: "unable to read identifier: a\"b, err = parse error on line 1, column 2: bare \" in non-quoted-field"},
+		{Input: `abc.cde`, Error: `unexpected number of parts 2 in identifier abc.cde, expected 1 in a form of "<account_object_name>"`},
+		{Input: `""`, Expected: sdk.NewAccountObjectIdentifier(``)},
+		{Input: `""""`, Expected: sdk.NewAccountObjectIdentifier(`"`)},
+		{Input: `abc`, Expected: sdk.NewAccountObjectIdentifier(`abc`)},
+		{Input: `"abc"`, Expected: sdk.NewAccountObjectIdentifier(`abc`)},
+		{Input: `"ab.c"`, Expected: sdk.NewAccountObjectIdentifier(`ab.c`)},
+		{Input: `"a""bc"`, Expected: sdk.NewAccountObjectIdentifier(`a"bc`)},
```

Comment:
I would ask about the uppercase but this is of course pointless in the unit tests because they will work the same. However, it would be great to have integration tests for these parsers, i.e. let's take a few object that have id fields in show/describe, create the object with one of the above id, and check if we can parse it from the response correctly (and there the casing may matter)

Reply:
Added integration tests, but more to document the Snowflake identifier behavior for certain scenarios.